src/pytesmo/validation_framework/data_averaging.py

```diff
@@ -0,0 +1,430 @@
+    def lut(self) -> dict:
+        """Get a lookup table that combines the points falling under the same reference pixel"""
+        lut = {}
+        for other_name, other_class in self.others_class.items():
+            try:
+                grid = other_class.grid
+            except AttributeError:
+                # when filters are applied, this will be an AdvancedMaskingAdapter
+                grid = other_class.cls.cls.grid
+            # subsetting shouldn't be necessary with ismn reader, but for satellites
+            other_points = grid.get_bbox_grid_points(
+                *self.geo_subset,
+                both=True,
+            )
+            other_lut = {}
+            # iterate from the side of the non-reference
+            for gpi, lat, lon in zip(other_points[0], other_points[1], other_points[2]):
+                # list all non-ref points under the same ref gpi
+                try:
+                    ref_gpi = self.ref_class.grid.find_nearest_gpi(lon, lat)[0]
+                except AttributeError:
+                    ref_gpi = self.ref_class.cls.cls.grid.find_nearest_gpi(lon, lat)[0]
+                if ref_gpi in other_lut.keys():
+                    other_lut[ref_gpi].append((gpi, lon, lat))
+                else:
+                    other_lut[ref_gpi] = [(gpi, lon, lat)]
+            # add to dictionary even when empty
+            lut[other_name] = other_lut
+
+        return lut
+
+    def get_timeseries(
+            self,
+            points,
+            other_name,
+    ) -> pd.DataFrame:
+        """
+        Get the timeseries for given points info and return them in a list.
+
+        Parameters
+        ----------
+        points: list of tuples
+            list of tuples of (gpi, lon, lat)
+        other_name: str
+            Name of the dataset which the points belong to
+
+        Returns
+        -------
+        dss: list
+            list of dataframes of the reference timeseries
+        """
+        dss = []
+        for gpi, lon, lat in points:
+            ds = self.read_ds(other_name, gpi)
+            dss.append(ds)
+
+        return dss
+
+    @staticmethod
+    def temp_match(
+            to_match,
+            method='rescale',
+            hours=6,
+            **kwargs
+    ) -> pd.DataFrame:
+        """
+        Temporal match by:
+            * temporal matching to the longest timeseries
+            * taking the common values only
+
+        Parameters
+        ----------
+        to_match: list
+            list of dataframes to match
+        method: str
+            matching method. Either 'common' or 'rescale'
+        hours: int
+            window to perform the temporal matching
+
+        Returns
+        -------
+        matched: pd.DataFrame
+            dataframe with temporally matched timeseries
+        """
+        if method == 'common':
+            matched = pd.concat(to_match, axis=1, join="inner").dropna(how="any")
+
+        elif method == 'rescale':
```

Comment:
I think 'rescale' is a bit confusing here, because in pytesmo terminology, rescaling is the stuff that's done in the scaling module (CDF matching, min-max-scaling, mean-std-scaling

Reply:
That's true. anyways, I have decided that it doesn't make sense to have both methods and I'll only keep the temporal matching option